packages/web-ui/src/ds/molecules/Chat/ChatTextArea/index.tsx

```diff
@@ -0,0 +1,55 @@
+'use client'
+
+import { KeyboardEvent, useCallback, useState } from 'react'
+
+import { Button } from '$ui/ds/atoms'
+import TextareaAutosize from 'react-textarea-autosize'
+
+export function ChatTextArea({
+  placeholder,
+  disabled = false,
+  onSubmit,
+}: {
+  placeholder: string
+  disabled?: boolean
+  onSubmit?: (value: string) => void
```

Comment:
Is it always gonna be the same onsubmit or do u expect several consumers to pass different implementations of onsubmit? if it's the former give it a more meaningful name.

function props are a tradeoff because they introduce indirection, make it easier for the reader to understand what this will do when possible.